compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -549,7 +549,7 @@ class TypeErasure(isJava: Boolean, semiEraseVCs: Boolean, isConstructor: Boolean
         if (inst.exists) sigName(inst) else tpnme.Uninstantiated
       case tp: TypeProxy =>
         sigName(tp.underlying)
-      case _: ErrorType | WildcardType =>
+      case _: ErrorType | WildcardType | NoType =>
         tpnme.WILDCARD
       case tp: WildcardType =>
         sigName(tp.optBounds)
```

Comment:
Does this code match _twice_ against `WildcardType`? I don't get it. In fact, the latter matches appear to be erased away:

```
scala> def foo: Any => Any = { case _: String | Int => 1; case _: Int => 2 }
def foo: Any => Any
scala> foo(1)
val res0: Any = 2
scala> def foo: Any => Any = { case _: String | AnyRef => 1; case _: AnyRef => 2 }
def foo: Any => Any
scala> foo(new AnyRef())
val res1: Any = 2
```

Reply:
It first matches against the object `WildcardType`, then the case below matches against any instance of the class `WildcardType`.

Reply:
Yeah sorry, Allan explained me what's going on. But do we really know that the only instance of `WildcardType(NoType)` is the `WildcardType` object?